cmd/create/admin/cmd.go

```diff
@@ -0,0 +1,223 @@
+				"must contain only letters, digits, dashes and underscores",
+			clusterKey,
+		)
+		os.Exit(1)
+	}
+
+	reporter.Warnf("It is recommended to add an identity provider to login to this cluster. " +
+		"See 'moactl create idp --help' for more information.")
+
+	// Create the AWS client:
+	awsClient, err := aws.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create AWS client: %v", err)
+		os.Exit(1)
+	}
+
+	awsCreator, err := awsClient.GetCreator()
+	if err != nil {
+		reporter.Errorf("Failed to get AWS creator: %v", err)
+		os.Exit(1)
+	}
+
+	// Create the client for the OCM API:
+	ocmConnection, err := ocm.NewConnection().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create OCM connection: %v", err)
+		os.Exit(1)
+	}
+	defer func() {
+		err = ocmConnection.Close()
+		if err != nil {
+			reporter.Errorf("Failed to close OCM connection: %v", err)
+		}
+	}()
+
+	// Get the client for the OCM collection of clusters:
+	clustersCollection := ocmConnection.ClustersMgmt().V1().Clusters()
+
+	// Try to find the cluster:
+	reporter.Debugf("Loading cluster '%s'", clusterKey)
+	cluster, err := ocm.GetCluster(clustersCollection, clusterKey, awsCreator.ARN)
+	if err != nil {
+		reporter.Errorf("Failed to get cluster '%s': %v", clusterKey, err)
+		os.Exit(1)
+	}
+
+	if cluster.State() != cmv1.ClusterStateReady {
+		reporter.Errorf("Cluster '%s' is not yet ready", clusterKey)
+		os.Exit(1)
+	}
+
+	// TODO: Verify that the htpasswd IdP does not already exist
+
+	// TODO: Verify that the user does not already exist
+
+	password, err := generateRandomPassword(23)
+	if err != nil {
+		reporter.Errorf("Failed to generate a random password")
+		os.Exit(1)
+	}
+
+	// Add admin user to the cluster-admins group:
+	reporter.Debugf("Adding '%s' user to cluster '%s'", username, clusterKey)
+	user, err := cmv1.NewUser().ID(username).Build()
+	if err != nil {
+		reporter.Errorf("Failed to create user '%s' for cluster '%s'", username, clusterKey)
+		os.Exit(1)
+	}
+	_, err = clustersCollection.Cluster(cluster.ID()).
+		Groups().Group("cluster-admins").
+		Users().Add().Body(user).
+		Send()
+	if err != nil {
+		reporter.Errorf("Failed to add user '%s' to cluster '%s': %v", username, clusterKey, err)
+		os.Exit(1)
+	}
+
+	// Create HTPasswd IDP configuration:
+	reporter.Debugf("Adding '%s' udp to cluster '%s'", idpName, clusterKey)
+	htpasswdIDP := cmv1.NewHTPasswdIdentityProvider().
+		Username(username).
+		Password(password)
+
+	// Create new IDP with HTPasswd provider:
+	idp, err := cmv1.NewIdentityProvider().
+		Type("HTPasswdIdentityProvider"). // FIXME: ocm-api-model has the wrong enum values
+		Name(idpName).
+		MappingMethod(cmv1.IdentityProviderMappingMethod("claim")).
+		Htpasswd(htpasswdIDP).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create '%s' configuration for cluster '%s'", idpName, clusterKey)
```

Comment:
configuration should be identity provider

Reply:
Done, changed it in other places too